plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,303 @@
+"""Functionality for determining the floating potential of a Langmuir sweep."""
+__all__ = ["find_floating_potential"]
+
+import numpy as np
+
+from collections import namedtuple
+from warnings import warn
+from typing import Union
+
+from plasmapy.analysis import fit_functions as ffuncs
+
+FloatingPotentialResults = namedtuple(
+    "FloatingPotentialResults",
+    ("vf", "vf_err", "rsq", "func", "islands", "indices"),
+)
```

Comment:
Noted.  While slower in creation than `NamedTuple`, `Dataclass` does beat `NamedTuple` in all other metrics.  However, we have not officially dropped Python 3.6, so I think it should stay a `NamedTuple`.

Reply:
Numpy's NEP 37 or whichever it was does suggest dropping 3.6 by now iirc. Since we depend on Numpy, we're softly tied to their release schedule, even if we haven't officially dropped 3.6.

Reply:
I like the concept of a `Dataclass`, but I still think a `NamedTuple` is more appropriate in this use case. (1) You can still index a named tuple and I can see users just wanting to take the first element and running with it.  I see all the swept Langmuir analysis functionality having a return similar to this, so users can always get the first element by doing `results[0]` versus `results.vf`, `results.vp`, `results.te`, etc. (2) Function returns of multiple values are always a tuple, so a `NamedTuple` is more inline with what is expected. (3)  I actually got this idea from `numpy` so, if it's good for them, it's good for me.